# HG changeset patch
# User Francois Bissey <francois.bissey@canterbury.ac.nz>
# Date 1323603049 -46800
# Node ID 27b676bc8b4d15d20aa3792b7dc698e0031b2e06
# Parent  df5620969fbddef0cf191407d046c2ed522d6a49
trac 9958: it.next() successive values are in random order, this is really apparent with python 2.7.

diff --git a/sage/combinat/words/nfactor_enumerable_word.py b/sage/combinat/words/nfactor_enumerable_word.py
--- a/sage/combinat/words/nfactor_enumerable_word.py
+++ b/sage/combinat/words/nfactor_enumerable_word.py
@@ -13,15 +13,18 @@
 
 EXAMPLES:
 
-Enumeration of factors::
+Enumeration of factors, the successive values returned by it.next()                                                                          
+can appear in a different order depending on hardware. Therefore we                                                                          
+mark the three first results of the test random. The important test                                                                          
+is that the iteration stops properly on the fourth call::
 
     sage: w = Word([4,5,6])^7
     sage: it = w.factor_iterator(4)
-    sage: it.next()
+    sage: it.next() # random
     word: 6456
-    sage: it.next()
+    sage: it.next() # random
     word: 5645
-    sage: it.next()
+    sage: it.next() # random
     word: 4564
     sage: it.next()
     Traceback (most recent call last):
